src/components/TodoItem.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useCallback } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { deleteTodos, editTodos } from '../api/todos';
+import { useTodosContext } from './useTodosContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    list,
+    setList,
+    editingTodo,
+    setEditingTodo,
+    handleError,
+    setLoadingTodosIds,
+    setErrorMessage,
+    loadingTodosIds,
+  } = useTodosContext();
+
+  const handleCheckbox = (checkedTodo: Todo) => {
+    const updatedList = list.map((item: Todo) => {
+      if (item.id === checkedTodo.id) {
+        return { ...item, completed: !item.completed };
+      }
+
+      return item;
+    });
+
+    setList(updatedList);
+  };
+
+  const handleTodoDelete = useCallback(
+    (todoId: number) => {
+      deleteTodos(todoId)
+        .then(() => {
+          setList(currentTodos => {
+            return currentTodos.filter(
+              currentTodo => currentTodo.id !== todoId,
+            );
+          });
+          setLoadingTodosIds([]);
+        })
+        .catch(() => {
+          setErrorMessage('Unable to delete a todo');
+          setLoadingTodosIds([]);
```

Comment:
- Remove loaders in the finally block as you do it in both cases
- Use `handleError` function

Reply:
dont have finally here at all